0002-chunking.md

```diff
@@ -0,0 +1,271 @@
+
+The `Pickles.compile` function must be modified to check for the number of chunks in each inductive rule's previous rule tags. When these numbers are not consistent between inductive rules, the code must throw an error in order to achieve layout homogeneity.
+
+### Prototype PRs
+
+* [Fix chunking for proofs](https://github.com/o1-labs/proof-systems/pull/1032)
+* [Update bindings for chunking fix](https://github.com/MinaProtocol/mina/pull/13285)
+* [Enable chunked proving](https://github.com/o1-labs/proof-systems/pull/1033)
+* [Add zero-knowledge for chunked proofs](https://github.com/o1-labs/proof-systems/pull/1045)
+* [Update bindings to enable chunking](https://github.com/MinaProtocol/mina/pull/13286)
+
+## Test Plan and Functional Requirements
+[test-plan-and-functional-requirements]: #test-plan-and-functional-requirements
+
+1. Testing Goals and Objectives:
+  * A SnarkyJS proof is able to achieve use more than the previous circuit size limit.
+  * A SnarkyJS proof using more than the previous circuit size limit is able to be applied in a zkApp transaction by Berkeley.
+2. Testing Approach:
+  * Unit testing, at each of the kimchi, pickles, and SnarkyJS layers.
+3. Testing Scope:
+  * That proofs are produced, and that they verify, in kimchi, pickles, SnarkyJS, and Berkeley.
+4. Testing Requirements:
+5. Testing Resources:
+  * Additional CI time on each PR to proof-systems, mina, and SnarkyJS.
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+The existing recursion layer in pickles allows most zkApps that would hit the limit to be restructured in a way that doesn't. However, to do this requires significant extra expertise.
+
+## Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+A previous alternative was to increasing the circuit size limit by generating a larger (`2^17`) URS. This was rejected after the slow-down for verifying kimchi proofs on Berkeley was deemed unacceptable.
+
+Since this feature was already partly implemented, it is significantly lower-cost than other potential proof-system changes.
+
+None of the other alternatives were able to increase the circuit size limit without either breaking compatibility with zkApps or dramatically increasing complexity and proving time. In particular, switching the polynomial commitments scheme, and using a different proof system were considered and rejected by the crypto team.
+
+Failure to complete this feature will result in less usability for zkApps, which will have a negative effect on both the Berkeley hard fork and SnarkyJS.
+
+## Prior art
+[prior-art]: #prior-art
+
+The existing broken chunking implementation.
+
+## Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+* What parts of the design do you expect to resolve through the RFC process before this gets merged?
+  - SnarkyJS design
```

Comment:
What exactly do we need to expose to the user? The number of chunks? Can't we compute that at circuit compile time? This is important for the SnarkyJS API design